nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/PutMarkLogicRecord.java

```diff
@@ -154,15 +157,14 @@ public final void onTrigger(ProcessContext context, ProcessSession session) thro
             Record record;
             while ((record = reader.nextRecord()) != null) {
                 baos.reset();
-                Map<String, String> additionalAttributes = Collections.emptyMap();
                 try (final RecordSetWriter writer = writerFactory.createWriter(getLogger(), schema, baos)) {
-                    final WriteResult writeResult = writer.write(record);
-                    additionalAttributes = writeResult.getAttributes();
+                    writer.write(record);
                     writer.flush();
                     BytesHandle bytesHandle = new BytesHandle().with(baos.toByteArray());
                     final String uriKey = uriFieldName == null ? UUID.randomUUID().toString() : record.getAsString(uriFieldName);
-                    WriteEvent writeEvent = buildWriteEvent(context, session, flowFile, uriKey, bytesHandle, additionalAttributes);
+                    WriteEvent writeEvent = buildWriteEvent(context, session, flowFile, uriKey, bytesHandle, metadata);
                     this.addWriteEvent(writeBatcher, writeEvent);
+                    added++;
```

Comment:
This fixes the issue with zero being reported to the provenance report every time.